src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -462,7 +600,31 @@ private void onSignal(
                 {
                     writer.doAbort(target, targetRouteId, targetStreamId, traceId, targetAuthorization);
                 }
+            }
```

Comment:
Replace this with a switch on `signalId` and a `case TOKEN_EXPIRED_SIGNAL:` that delegates to `onTokenExpiredSignal(SignalFW)`.

Then in `onTokenExpiredSignal(SignalFW)` put the body of the if block above, and inline the body of `tryExtendGrantExpiration()` too.